src/app/Dashboard/AddCard.tsx

```diff
@@ -157,3 +227,169 @@ export const AddCard: React.FunctionComponent<AddCardProps> = (props: AddCardPro
+              onPlus={handleNumericStep(ctrl.key, 1)}
+              onMinus={handleNumericStep(ctrl.key, -1)}
+            />
+          );
+          break;
+        case 'string':
+          input = (
+            <TextInput
+              type="text"
+              aria-label={`${ctrl.key} input`}
+              value={props.config[ctrl.key]}
+              onChange={handleChange(ctrl.key)}
+            />
+          );
+          break;
+        case 'text':
+          input = (
+            <TextArea
+              type="text"
+              aria-label={`${ctrl.key} input`}
+              value={props.config[ctrl.key]}
+              onChange={handleChange(ctrl.key)}
+            />
+          );
+          break;
+        case 'select':
+          input = <SelectControl handleChange={handleChange(ctrl.key)} config={props.config} control={ctrl} />;
+          break;
+        default:
+          input = <Text>Bad config</Text>;
+          break;
+      }
+      return (
+        <FormGroup key={`${ctrl.key}}`} label={ctrl.name} helperText={ctrl.description} isInline isStack>
+          {input}
+        </FormGroup>
+      );
+    },
+    [props.config, handleChange, handleNumeric, handleNumericStep]
+  );
+
+  return (
+    <>
+      {props.controls.length > 0 ? (
+        <Form>
+          <Title headingLevel={'h5'}>Configure the {props.cardTitle} card</Title>
+          {props.controls.map((ctrl) => createControl(ctrl))}
+        </Form>
+      ) : (
+        <Text>No configuration required.</Text>
+      )}
+    </>
+  );
+};
+
+const SelectControl = (props: { handleChange: ({}) => void; control: PropControl; config: any }) => {
+  const addSubscription = useSubscriptions();
+
+  const [selectOpen, setSelectOpen] = React.useState(false);
+  const [options, setOptions] = React.useState([] as string[]);
+  const [errored, setErrored] = React.useState(false);
+
+  const handleSelect = React.useCallback(
+    (_, selection, isPlaceholder) => {
+      if (!isPlaceholder) {
+        props.handleChange(selection);
+      }
+      setSelectOpen(false);
+    },
+    [props.handleChange, setSelectOpen]
+  );
+
+  React.useEffect(() => {
+    let obs;
+    if (props.control.values instanceof Observable) {
+      obs = props.control.values;
+    } else {
+      obs = of(props.control.values);
+    }
+    addSubscription(
+      obs.subscribe({
+        next: (v) => {
+          setErrored(false);
+          setOptions((old) => {
+            if (Array.isArray(v)) {
+              return v;
+            }
+            return [...old, v];
+          });
+        },
+        error: (err) => {
+          setErrored(true);
+          setOptions([err]);
+        },
+      })
+    );
+  }, [props.control, props.control.values, of, addSubscription, setOptions, setErrored]);
+
+  return (
+    <Select
+      onToggle={setSelectOpen}
+      isOpen={selectOpen}
+      onSelect={handleSelect}
+      selections={props.config[props.control.key]}
```

Comment:
Should this take into account error? 

For example, previously users choose a valid selection (i.e. still fetching options) - async0. But later the fetch failed but the selected holder is till the previously valid one (async0) tho the list of option only contains error.

Reply:
If so, we might need to do clean up current propConfig to reset to default.

Reply:
I think regardless of attempts to clean up, if an error occurs when populating options on this form then the result is going to be an invalid card configuration. If anything the wizard should probably lock and prevent the user from moving forward, so they have to cancel out and clear the state that way.

Reply:
That probably also ties in to a need for this wizard to allow form configurations that have required fields and validations too, and if any fields are unfilled/invalid/erorred then the wizard should lock. I think a higher priority than adding all this nice error handling and cleanup is to focus instead on adding card implementations that can even generate these invalid/incomplete configs and errors ;-)

Reply:
Ah right makes sense :D We can refactor and add these in the future.